pkg/apis/monitoring/v1/types.go

```diff
@@ -125,14 +125,19 @@ type CommonPrometheusFields struct {
 	// _not_ be added when value is set to empty string (`""`).
 	PrometheusExternalLabelName *string `json:"prometheusExternalLabelName,omitempty"`
 	// Log level for Prometheus to be configured with.
+	//+kubebuilder:validation:Enum=debug;info;warn;error
 	LogLevel string `json:"logLevel,omitempty"`
 	// Log format for Prometheus to be configured with.
+	//+kubebuilder:validation:Enum=logfmt;json
 	LogFormat string `json:"logFormat,omitempty"`
 	// Interval between consecutive scrapes. Default: `1m`
+	// +kubebuilder:validation:Pattern:="^[0-9]+(ms|s|m|h|d|w|y)$"
```

Comment:
model.Duration allows a value of 0 (from [here](https://github.com/prometheus/common/blob/0e1254b764d0c37ae98f7b4b130b22c42844c92d/model/time.go#L192-L194)) as well as more complex durations (e.g. `1m2s`).

```suggestion
	// +kubebuilder:validation:Pattern:="^0|(([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?$"
	// +kubebuilder:default:=30s
```